chain/consensus/hierarchical/subnet/resolver/resolver.go

```diff
@@ -0,0 +1,499 @@
+package resolver
```

Comment:
@Lefkok, and this is the implementation of the cross-messages resolution protocol (the one that pull-push cross-messages from other subnets). In case you want to also look at it (this is less of a priority than the previous comment).